### ISSUE TITLE
Docs: Are there outdated APIs (e.g., ServletWebServerFactoryCustomizer) in the docs?

### ISSUE THREAD

**Issue by saleese** [spring-projects/spring-boot#11529]
I am checking the APIs in the spring-boot-docs folder(https://github.com/spring-projects/spring-boot/tree/master/spring-boot-project/spring-boot-docs/src/main/asciidoc), and I found outdated APIs. I hope some contributors may confirm whether the followings are outdated or not:

1. "ServletWebServerFactoryCustomizer" should be replaced by "WebServerFactoryCustomizer" in https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-docs/src/main/asciidoc/howto.adoc. 

2. "EmbeddedWebServer" and "SpringBootServletInitializer" look to have been deleted but still exists in the same document "howto.adoc."

3. "EnableIntegrationPatterns" is outdated in https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-docs/src/main/asciidoc/spring-boot-cli.adoc.

4. "YamlMapFactoryBean" is outdated https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-docs/src/main/asciidoc/spring-boot-features.adoc.

Thanks in advance,
Seonah

**Comment by philwebb** [comment]
Looks like we missed these during the big web stack refactor.

**Comment by snicoll** [comment]
Thanks for that list. FYI, `SpringBootServletInitializer` hasn't been removed.

**Comment by snicoll** [comment]
@saleese I haven't found that `YamlMapFactoryBean` is outdated. If you can provide more feedback as why you think it is, I am happy to revisit that part of the doc. Thanks!

**Comment by saleese** [comment]
YamlMapFactoryBean was mis-reported. I analyzed the revision histories of the spring-boot, and found that it was deleted from spring-boot in July 2014. Now, I find that it was moved to the spring-beans project by checking the link: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/YamlMapFactoryBean.html. I should have checked it before my report.
